scripts/omni_run.py

```diff
@@ -2,220 +2,184 @@
-    uv.zenith_dec = np.array([aa.lat] * uv.Nblts)
-
-    # antenna information
-    uv.Nants_telescope = 128
-    uv.antenna_numbers = np.arange(uv.Nants_telescope, dtype=int)
-    uv.Nants_data = len(np.unique(np.concatenate([uv.ant_1_array, uv.ant_2_array]).flatten()))
-    uv.antenna_names = np.array(['ant{0}'.format(ant) for ant in uv.antenna_numbers])
-    antpos = []
-    for k in aa:
-        antpos.append(k.pos)
+args = np.sort(args)
+pols = opts.pol.split(',')
 
-    uv.antenna_positions = np.array(antpos)
+if len(args)==0: raise AssertionError('Please provide visibility files.')
+if opts.minV and len(list(set(''.join(pols))))==1: raise AssertionError('Stokes V minimization requires crosspols in the "-p" option.')
 
-    if returnuv:
-        return uv
-    else:
-        print('   Saving {0}'.format(filename))
-        uv.write_uvfits(filename, force_phase=True, spoof_nonessential=True)
+def getPol(fname): return fname.split('.')[3] #XXX assumes file naming format
+def linPol(polstr): return len(list(set(polstr)))==1
 
+linear_pol_keys = []
+for pp in pols:
+    if linPol(pp):
+        linear_pol_keys.append(pp)
 
-# Dictionary of calpar gains and files
-pols = opts.pol.split(',')
-files = {}
-if not opts.firstcal:
-    raise ValueError('Please provide a firstcal file. Exiting...')
-else:
-    firstcal_files = np.sort(glob.glob(opts.firstcal))
-for f, filename in enumerate(args):
-    files[filename] = {}
-    if len(firstcal_files) == len(args):
-        files[filename]['firstcal'] = str(firstcal_files[f])  # one firstcal file per input file.
-    elif len(args) > len(firstcal_files):
-        files[filename]['firstcal'] = str(firstcal_files[0])  # only use first firstcal file for all input files.
-    for p in pols:
-        fn = filename.split('.')
-        fn[3] = p
-        files[filename][p] = '.'.join(fn)
-    
 # Create info
 # generate reds from calfile
 aa = aipy.cal.get_aa(opts.cal, np.array([.15]))
 print 'Getting reds from calfile'
 if opts.ex_ants:  # assumes exclusion of the same antennas for every pol
-    ex_ants = []
-    for a in opts.ex_ants.split(','):
-        ex_ants.append(int(a))
+    ex_ants = map(int,opts.ex_ants.split(','))
     print '   Excluding antennas:', sorted(ex_ants)
 else:
     ex_ants = []
-info = aa_to_info(aa, pols=list(set(''.join(pols))), ex_ants=ex_ants, crosspols=pols)
+info = aa_to_info(aa, pols=list(set(''.join(pols))), ex_ants=ex_ants, crosspols=pols, minV=opts.minV)
 reds = info.get_reds()
 bls = [bl for red in reds for bl in red]
 
-### Omnical-ing! Loop Through Files ###
-for f, filename in enumerate(args):
-    file_group = files[filename]  # dictionary with pol indexed files
-    print 'Reading:'
-    for key in file_group.keys():
-        print '   ' + file_group[key]
+### Collect all firstcal files ###
+firstcal_files = {}
+if not opts.firstcal:
+    raise ValueError('Please provide a firstcal file. Exiting...') #XXX this requires a firstcal file for any implementation
+else:
```

Comment:
No need to put this code inside an `else` clause — if the `if` test evaluates to true, the program's over.